src/client/connect/http.rs

```diff
@@ -159,14 +161,17 @@ impl<R> HttpConnector<R> {
         self.config_mut().recv_buffer_size = size;
     }
 
-    /// Set that all sockets are bound to the configured address before connection.
+    /// Set that all sockets are bound to the configured addresses before connection.
     ///
     /// If `None`, the sockets will not be bound.
     ///
     /// Default is `None`.
     #[inline]
-    pub fn set_local_address(&mut self, addr: Option<IpAddr>) {
-        self.config_mut().local_address = addr;
+    pub fn set_local_address(&mut self, addr_ipv4: Option<Ipv4Addr>, addr_ipv6: Option<Ipv6Addr>) {
```

Comment:
This would be a breaking change. To prevent that we could do:

- Keep `set_local_address(Option<IpAddr>)`, and inside match on if it is v4 or v6.
- Add `set_local_address_v6(Option<Ipv6Addr>)`.

Reply:
What's your preference here? I would be happy with either.

Reply:
Oh right, I didn't think of that... 

```rust
connector.set_local_address(Some(v4_addr));
connector.set_local_address(Some(v6_addr));
```

That could be the way to set both... Hm. Could be nice, as long as it's not confusing (would a user assume it would have *overwritten* the previous value?).

Reply:
It could be confusing: such API implies that arbitrary number of egress IPs can be added. However, in reality we only choose between the two.  And `set_` prefix usually has override semantics when such methods used multiple times. 

I guess the best option would be to introduce something like `set_locall_address_versioned(Ipv4Addr, Ipv6Addr)` (note that arguments are not optional). So, if it's a singular value of either version one would use `set_local_address` and if they want to specify both versions they would use `set_locall_address_versioned`

Reply:
Yea I agree with what you said. Let's make it a new method, without options. Another name option is `set_local_addresses`, plural. :shrug: